vsutil/__init__.py

```diff
@@ -279,3 +272,56 @@ def _resolve_enum(enum: Type[E], value: Any, var_name: str, module: Optional[str
         return enum(value)
     except ValueError:
         raise ValueError(f'{var_name} must be in {_readable_enums(enum, module)}.') from None
+
+
+def _should_dither(out_bits: int,
+                   out_range: Optional[Range] = None,
+                   out_sample_type: Optional[vs.SampleType] = None,
+                   clip: Optional[vs.VideoNode] = None,
+                   in_bits: Optional[int] = None,
+                   in_range: Optional[Range] = None,
+                   in_sample_type: Optional[vs.SampleType] = None,
+                   ) -> bool:
+    """
+    Determines whether dithering is needed for a given depth/range/sample_type conversion.
+
+    If a clip is given, the clip's properties will take precedence over the in_bits and in_sample_type parameters.
+    If a variable-format clip is given, in_bits and in_sample_type are required.
```

Comment:
This is an internal helper function that is only used once and will likely never be used elsewhere.
It’s not necessary to have multiple ways of passing parameters. It’s just needlessly complicated.